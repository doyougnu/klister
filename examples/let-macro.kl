#lang kernel

(import (shift kernel 1))
(import (shift "prim-syntax.kl" 1))

(example
  (let-syntax
    (let1 (lambda (stx)
      (syntax-case stx
        ((list (_ binder body))
          (syntax-case binder
            ((list (x e))
              (pure (list-syntax
                ((list-syntax
                  ((ident-syntax 'lambda stx)
                    (list-syntax (x) stx)
                      body) stx) e) stx))))))))
              (let1 (x (lambda (x) x))
                x)))
