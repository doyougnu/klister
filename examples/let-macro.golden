Unknown: #[let-macro.kl:5.12-5.18]<lambda>
